src/HLS/module_generator/ReadWrite_m_axiModuleGenerator.cpp

```diff
@@ -177,27 +177,15 @@ void ReadWrite_m_axiModuleGenerator::InternalExec(std::ostream& out, structural_
    out << "localparam BITSIZE_address=BITSIZE_" << _ports_in[i_in4].name << ",\n"
        << "  BITSIZE_bus=" << _ports_out[o_wdata].type_size << ",\n"
        << "  BITSIZE_bus_size=BITSIZE_bus/8,\n"
-       << "  BITSIZE_data=BITSIZE_" << _ports_in[i_in3].name << ",\n"
+       << "  BITSIZE_data=" << _ports_in[i_in3].type_size << ",\n"
        << "  BITSIZE_data_size=BITSIZE_data/8,\n"
+       << "  BITSIZE_log_data_size=" << ceil_log2(_ports_in[i_in3].type_size / 8) << ",\n"
```

Comment:
Here it could be nice to add a comment to tell where the generated number comes from, so that it is still clear even if reading the Verilog alone.